### PR TITLE
[CGP-390] [Docs] Added a doc describing alternatives to elimination contexts

### PR DESCRIPTION
**Pull request by effectfully**

### CODE REVIEW

File: docs/fomega/deep-isorecursive/alternatives.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+Elimination contexts are problematic, see the problem.md document. Hence one option is to move away from them and use something else. This document describes possible alternatives.
+
+## Head-spine form
+
+An elimination context is either a type or an application of a type function (which is in the form of elimination context again) to a type argument. If the head of such application is a `fix`, then we can `wrap` and `unwrap` a term of such type. Wrapping results in the `head` of the application being substituted for the variable it binds. One of the premises `wrap` has is that the whole thing must be of kind `(type)` and the same holds for `unwrap`.
+
+What I've just described is exactly iterated application that leads to hereditary substitution at the type level: allowing `Fix` to be applied to a bunch of arguments kinda asks for iterated application at the type level. Imagine `Fix` is not a binder, then type-level application always appears in the rules, e.g. the conclusion of the `wrap` rule becomes
+
+```
+Gamma |- wrap α S M : [fix α S]
```
Comment (michaelpj) [inline docs/fomega/deep-isorecursive/alternatives.md]: Can you bracket the `fix alpha` for clarity?

Comment (effectfully) [reply docs/fomega/deep-isorecursive/alternatives.md]: Maybe just `Gamma |- wrap α S M : fix α S`?

Comment (michaelpj) [reply docs/fomega/deep-isorecursive/alternatives.md]: Then where's the iterated application?

Comment (effectfully) [reply docs/fomega/deep-isorecursive/alternatives.md]: Yep. Ok, I'll make it `[(fix α spine)]`.

Comment (michaelpj) [reply docs/fomega/deep-isorecursive/alternatives.md]: This makes me think I've completely misunderstood. Shouldn't it be `[(fix a) spine]` i.e. we're iteratively applying `(fix a)` to a spine of args?

Comment (effectfully) [reply docs/fomega/deep-isorecursive/alternatives.md]: Sorry, yes, you're right.

Comment (effectfully) [reply docs/fomega/deep-isorecursive/alternatives.md]: Oh well. That `S` is actually a type, not a spine. There is a sentence about generalizing `S` to be a spine. Ok, I reworded this and removed some confusing stuff.

**Pull request merged**